torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -58,8 +87,14 @@ void UnrollPass::handle(kir::Expr* expr) {
     if (!should_predicate) {
       return;
     }
+    auto thread_pred = getThreadPredicate(out_tv);
+    // TODO: better way to detect initialization of reduction buffers?
+    const bool is_reduction_init = out_tv->definition() != expr;
+    if (is_reduction_init) {
+      thread_pred = excludeGridReductionPred(thread_pred);
```

Comment:
If it is a reduction initialization isn't it always safe to turn off all thread predicates? Could we just simplify this with `thread_pred = is_reduction_init ? nullptr : getThreadPredicate(out_tv)` ?

I don't really like the logic of removing something we've computed, instead it seems this would be cleaner, or it would be cleaner to tell getThreadPredicate to not put grid reduction predicates in.

Reply:
I think you're right. Thanks for the feedback.